lib/rust/metamodel/src/meta/transform.rs

```diff
@@ -0,0 +1,110 @@
+//! Transformations on the meta representation.
+
+use crate::meta::*;
+use derivative::Derivative;
+
+
+
+// ===============
+// === Flatten ===
+// ===============
+
+/// `flatten` the specified fields into their containing structs, transitively.
+///
+/// Each inserted field will have its name prepended with the name of its eliminated container.
+/// If the `hide` property is set for the container, it will be inherited by its child fields.
+pub fn flatten(graph: &mut TypeGraph, ids: &mut BTreeSet<FieldId>) {
```

Comment:
This should have an example attached. It's not obvious what it does after reading the docs.

Reply:
This implements a transformation exposed through the `reflect` interface--I'll add an example to the docs for the `#[reflect(flatten)]` attribute, and link there to explain the motivation of this function.